# Comparing `tmp/pciSeq-0.0.52.dev1-py3-none-any.whl.zip` & `tmp/pciSeq-0.0.52rc0.dev0-py3-none-any.whl.zip`

## zipinfo {}

```diff
@@ -1,54 +1,55 @@
-Zip file size: 361109 bytes, number of entries: 72
--rw-rw-rw-  2.0 fat     2135 b- defN 23-Jun-25 12:12 pciSeq/__init__.py
--rw-rw-rw-  2.0 fat    10910 b- defN 23-Jun-25 12:17 pciSeq/app.py
--rw-rw-rw-  2.0 fat     3367 b- defN 23-Jun-24 12:16 pciSeq/config.py
+Zip file size: 364260 bytes, number of entries: 73
+-rw-rw-rw-  2.0 fat     2135 b- defN 23-Jun-25 20:54 pciSeq/__init__.py
+-rw-rw-rw-  2.0 fat    10910 b- defN 23-Jun-29 10:21 pciSeq/app.py
+-rw-rw-rw-  2.0 fat     3367 b- defN 23-Jun-29 10:21 pciSeq/config.py
 -rw-rw-rw-  2.0 fat     3678 b- defN 22-Apr-07 23:15 pciSeq/make_plot.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/__init__.py
--rw-rw-rw-  2.0 fat       27 b- defN 23-Jun-25 19:22 pciSeq/src/_version.py
+-rw-rw-rw-  2.0 fat       29 b- defN 23-Jul-02 21:05 pciSeq/src/_version.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/cell_call/__init__.py
--rw-rw-rw-  2.0 fat    20857 b- defN 23-Jun-25 12:05 pciSeq/src/cell_call/datatypes.py
+-rw-rw-rw-  2.0 fat    20864 b- defN 23-Jun-29 10:21 pciSeq/src/cell_call/datatypes.py
 -rw-rw-rw-  2.0 fat     2995 b- defN 23-Apr-26 22:13 pciSeq/src/cell_call/log_config.py
--rw-rw-rw-  2.0 fat    19153 b- defN 23-Jun-25 12:05 pciSeq/src/cell_call/main.py
--rw-rw-rw-  2.0 fat     3104 b- defN 23-Jun-25 12:05 pciSeq/src/cell_call/summary.py
--rw-rw-rw-  2.0 fat     9194 b- defN 23-Jun-25 12:11 pciSeq/src/cell_call/utils.py
+-rw-rw-rw-  2.0 fat    18989 b- defN 23-Jul-02 11:57 pciSeq/src/cell_call/main.py
+-rw-rw-rw-  2.0 fat     3104 b- defN 23-Jun-25 20:54 pciSeq/src/cell_call/summary.py
+-rw-rw-rw-  2.0 fat     9194 b- defN 23-Jun-25 20:54 pciSeq/src/cell_call/utils.py
 -rw-rw-rw-  2.0 fat        0 b- defN 23-Jun-11 12:10 pciSeq/src/diagnostics/__init__.py
 -rw-rw-rw-  2.0 fat      299 b- defN 23-Jun-24 12:16 pciSeq/src/diagnostics/config.py
--rw-rw-rw-  2.0 fat     3336 b- defN 23-Jun-25 12:05 pciSeq/src/diagnostics/diagnostics.py
--rw-rw-rw-  2.0 fat      796 b- defN 23-Jun-25 12:11 pciSeq/src/diagnostics/launch_diagnostics.py
--rw-rw-rw-  2.0 fat     7162 b- defN 23-Jun-25 12:11 pciSeq/src/diagnostics/utils.py
+-rw-rw-rw-  2.0 fat     3336 b- defN 23-Jun-25 20:54 pciSeq/src/diagnostics/diagnostics.py
+-rw-rw-rw-  2.0 fat      796 b- defN 23-Jun-25 20:54 pciSeq/src/diagnostics/launch_diagnostics.py
+-rw-rw-rw-  2.0 fat     7162 b- defN 23-Jun-25 20:54 pciSeq/src/diagnostics/utils.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Jan-19 17:53 pciSeq/src/preprocess/__init__.py
--rw-rw-rw-  2.0 fat    11253 b- defN 23-Jun-25 12:05 pciSeq/src/preprocess/cell_borders.py
+-rw-rw-rw-  2.0 fat    11253 b- defN 23-Jun-25 20:54 pciSeq/src/preprocess/cell_borders.py
+-rw-rw-rw-  2.0 fat     5323 b- defN 23-Jun-25 19:58 pciSeq/src/preprocess/regionprops.py
 -rw-rw-rw-  2.0 fat     6973 b- defN 23-Apr-26 22:13 pciSeq/src/preprocess/segmentation.py
--rw-rw-rw-  2.0 fat     4585 b- defN 23-Jun-25 12:05 pciSeq/src/preprocess/spot_labels.py
--rw-rw-rw-  2.0 fat       88 b- defN 23-Jun-24 12:16 pciSeq/src/preprocess/utils.py
+-rw-rw-rw-  2.0 fat     4585 b- defN 23-Jun-29 10:21 pciSeq/src/preprocess/spot_labels.py
+-rw-rw-rw-  2.0 fat       88 b- defN 23-Jun-29 10:21 pciSeq/src/preprocess/utils.py
 -rw-rw-rw-  2.0 fat        0 b- defN 22-Feb-20 14:12 pciSeq/src/viewer/__init__.py
--rw-rw-rw-  2.0 fat     2256 b- defN 23-Jun-24 12:16 pciSeq/src/viewer/run_flask.py
--rw-rw-rw-  2.0 fat     5007 b- defN 23-Jun-25 12:05 pciSeq/src/viewer/stage_image.py
--rw-rw-rw-  2.0 fat     7768 b- defN 23-Jun-25 15:45 pciSeq/src/viewer/utils.py
--rw-rw-rw-  2.0 fat    10895 b- defN 23-Jun-24 12:16 pciSeq/static/2D/index.html
+-rw-rw-rw-  2.0 fat     2256 b- defN 23-Jun-25 20:54 pciSeq/src/viewer/run_flask.py
+-rw-rw-rw-  2.0 fat     5007 b- defN 23-Jun-25 20:54 pciSeq/src/viewer/stage_image.py
+-rw-rw-rw-  2.0 fat     9109 b- defN 23-Jul-02 20:10 pciSeq/src/viewer/utils.py
+-rw-rw-rw-  2.0 fat    10895 b- defN 23-Jun-25 20:54 pciSeq/static/2D/index.html
 -rw-rw-rw-  2.0 fat     6506 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/genes_datatable.html
 -rw-rw-rw-  2.0 fat     9622 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/css/index.css
--rw-rw-rw-  2.0 fat     3135 b- defN 23-Jun-24 12:16 pciSeq/static/2D/viewer/css/progress.css
--rw-rw-rw-  2.0 fat     7435 b- defN 23-Jun-25 12:05 pciSeq/static/2D/viewer/js/classConfig.js
--rw-rw-rw-  2.0 fat     1474 b- defN 23-Jun-25 12:05 pciSeq/static/2D/viewer/js/config.js
+-rw-rw-rw-  2.0 fat     3135 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/css/progress.css
+-rw-rw-rw-  2.0 fat     7435 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/js/classConfig.js
+-rw-rw-rw-  2.0 fat     2314 b- defN 23-Jul-02 20:15 pciSeq/static/2D/viewer/js/config.js
 -rw-rw-rw-  2.0 fat     5589 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/customControl.js
--rw-rw-rw-  2.0 fat    22811 b- defN 23-Jun-25 12:05 pciSeq/static/2D/viewer/js/dapi.js
+-rw-rw-rw-  2.0 fat    23501 b- defN 23-Jul-02 19:25 pciSeq/static/2D/viewer/js/dapi.js
 -rw-rw-rw-  2.0 fat     5984 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/dataLoader.js
 -rw-rw-rw-  2.0 fat    17232 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/donut.js
 -rw-rw-rw-  2.0 fat     4106 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/dt.js
--rw-rw-rw-  2.0 fat     9835 b- defN 23-Jun-24 12:16 pciSeq/static/2D/viewer/js/glyphConfig.js
+-rw-rw-rw-  2.0 fat     9835 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/js/glyphConfig.js
 -rw-rw-rw-  2.0 fat     4389 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/glyphPaths.js
 -rw-rw-rw-  2.0 fat     7224 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/glyphs.js
--rw-rw-rw-  2.0 fat    12281 b- defN 23-Jun-24 12:16 pciSeq/static/2D/viewer/js/index.js
+-rw-rw-rw-  2.0 fat    12281 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/js/index.js
 -rw-rw-rw-  2.0 fat     1073 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/progress.js
 -rw-rw-rw-  2.0 fat     6367 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/stage_cells.js
--rw-rw-rw-  2.0 fat    12064 b- defN 23-Jun-24 12:16 pciSeq/static/2D/viewer/js/stage_glyphs.js
+-rw-rw-rw-  2.0 fat    12064 b- defN 23-Jun-25 20:54 pciSeq/static/2D/viewer/js/stage_glyphs.js
 -rw-rw-rw-  2.0 fat     9540 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/stage_markers.js
--rw-rw-rw-  2.0 fat     7534 b- defN 23-Jun-25 12:05 pciSeq/static/2D/viewer/js/stage_markers_patched.js
+-rw-rw-rw-  2.0 fat     7580 b- defN 23-Jul-02 20:05 pciSeq/static/2D/viewer/js/stage_markers_patched.js
 -rw-rw-rw-  2.0 fat    15523 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/stage_polygons.js
 -rw-rw-rw-  2.0 fat     4471 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/streaming-tsv-parser.js
 -rw-rw-rw-  2.0 fat     4692 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/viewerUtils.js
 -rw-rw-rw-  2.0 fat      995 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/L.Control.Layers.Tree.css
 -rw-rw-rw-  2.0 fat      421 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/Leaflet.Coordinates-0.1.3.css
 -rw-rw-rw-  2.0 fat   122544 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/bootstrap.min.css
 -rw-rw-rw-  2.0 fat     8669 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/css/index.css
@@ -61,14 +62,14 @@
 -rw-rw-rw-  2.0 fat    15374 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/leaflet-pip.js
 -rw-rw-rw-  2.0 fat     5927 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/leaflet.textpath.js
 -rw-rw-rw-  2.0 fat      265 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/preloader.js
 -rw-rw-rw-  2.0 fat     9097 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/MarkerContainer.js
 -rw-rw-rw-  2.0 fat     4434 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/bezier-easing.js
 -rw-rw-rw-  2.0 fat   613808 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js
 -rw-rw-rw-  2.0 fat    91874 b- defN 23-Apr-26 22:13 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js
--rw-rw-rw-  2.0 fat     1091 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/LICENCE
--rw-rw-rw-  2.0 fat     4116 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/METADATA
--rw-rw-rw-  2.0 fat       92 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/WHEEL
--rw-rw-rw-  2.0 fat       48 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/entry_points.txt
--rw-rw-rw-  2.0 fat        7 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/top_level.txt
-?rw-rw-r--  2.0 fat     6872 b- defN 23-Jun-25 19:22 pciSeq-0.0.52.dev1.dist-info/RECORD
-72 files, 1316828 bytes uncompressed, 349947 bytes compressed:  73.4%
+-rw-rw-rw-  2.0 fat     1091 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/LICENCE
+-rw-rw-rw-  2.0 fat     4119 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/METADATA
+-rw-rw-rw-  2.0 fat       92 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/WHEEL
+-rw-rw-rw-  2.0 fat       48 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/entry_points.txt
+-rw-rw-rw-  2.0 fat        7 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/top_level.txt
+?rw-rw-r--  2.0 fat     6983 b- defN 23-Jul-02 21:05 pciSeq-0.0.52rc0.dev0.dist-info/RECORD
+73 files, 1325027 bytes uncompressed, 352914 bytes compressed:  73.4%
```

## zipnote {}

```diff
@@ -51,14 +51,17 @@
 
 Filename: pciSeq/src/preprocess/__init__.py
 Comment: 
 
 Filename: pciSeq/src/preprocess/cell_borders.py
 Comment: 
 
+Filename: pciSeq/src/preprocess/regionprops.py
+Comment: 
+
 Filename: pciSeq/src/preprocess/segmentation.py
 Comment: 
 
 Filename: pciSeq/src/preprocess/spot_labels.py
 Comment: 
 
 Filename: pciSeq/src/preprocess/utils.py
@@ -192,26 +195,26 @@
 
 Filename: pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js
 Comment: 
 
 Filename: pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js
 Comment: 
 
-Filename: pciSeq-0.0.52.dev1.dist-info/LICENCE
+Filename: pciSeq-0.0.52rc0.dev0.dist-info/LICENCE
 Comment: 
 
-Filename: pciSeq-0.0.52.dev1.dist-info/METADATA
+Filename: pciSeq-0.0.52rc0.dev0.dist-info/METADATA
 Comment: 
 
-Filename: pciSeq-0.0.52.dev1.dist-info/WHEEL
+Filename: pciSeq-0.0.52rc0.dev0.dist-info/WHEEL
 Comment: 
 
-Filename: pciSeq-0.0.52.dev1.dist-info/entry_points.txt
+Filename: pciSeq-0.0.52rc0.dev0.dist-info/entry_points.txt
 Comment: 
 
-Filename: pciSeq-0.0.52.dev1.dist-info/top_level.txt
+Filename: pciSeq-0.0.52rc0.dev0.dist-info/top_level.txt
 Comment: 
 
-Filename: pciSeq-0.0.52.dev1.dist-info/RECORD
+Filename: pciSeq-0.0.52rc0.dev0.dist-info/RECORD
 Comment: 
 
 Zip file comment:
```

## pciSeq/src/_version.py

```diff
@@ -1 +1 @@
-__version__ = '0.0.52.dev1'
+__version__ = '0.0.52.rc.dev'
```

## pciSeq/src/cell_call/datatypes.py

```diff
@@ -245,18 +245,18 @@
 
     def ini_cellProb(self, neighbors, cfg):
         nS = self.data.shape[0]
         nN = cfg['nNeighbors'] + 1
         SpotInCell = self.data.label
         # assert (np.all(SpotInCell.index == neighbors.index))
 
-        # sanity check (this actually needs to be rewritten)
-        mask = np.greater(SpotInCell, 0, where=~np.isnan(SpotInCell))
-        sanity_check = neighbors[mask, 0] + 1 == SpotInCell[mask]
-        assert ~any(sanity_check), "a spot is in a cell not closest neighbor!"
+        ## sanity check (this actually needs to be rewritten)
+        # mask = np.greater(SpotInCell, 0, where=~np.isnan(SpotInCell))
+        # sanity_check = neighbors[mask, 0] + 1 == SpotInCell[mask]
+        # assert ~any(sanity_check), "a spot is in a cell not closest neighbor!"
 
         pSpotNeighb = np.zeros([nS, nN])
         pSpotNeighb[neighbors == SpotInCell.values[:, None]] = 1
         pSpotNeighb[SpotInCell == 0, -1] = 1
 
         ## Add a couple of checks here
         return pSpotNeighb
```

## pciSeq/src/cell_call/main.py

```diff
@@ -85,17 +85,15 @@
             self.has_converged, delta = utils.hasConverged(self.spots, p0, self.config['CellCallTolerance'])
             logger.info(' Iteration %d, mean prob change %f' % (i, delta))
 
             # replace p0 with the latest probabilities
             p0 = self.spots.parent_cell_prob
 
             if self.config['is_redis_running']:
-                logger.info('start db publish')
                 self.redis_upd()
-                logger.info('end db publish')
 
             if self.has_converged:
                 # self.counts_within_radius(20)
                 iss_df, gene_df = collect_data(self.cells, self.spots, self.genes, self.single_cell)
                 break
 
             if i == max_iter-1:
@@ -377,15 +375,14 @@
         df = pd.merge(temp, df, right_index=True, left_on='cell_label', how='left')
 
         # ignore the first row, it is the background
         return df.iloc[1:, :]
 
     # -------------------------------------------------------------------- #
     def redis_upd(self):
-        logger.info("redis start")
         eta_bar_df = pd.DataFrame({
             'gene_efficiency': self.genes.eta_bar,
             'gene': self.genes.gene_panel
         })
         self.redis_db.to_redis(eta_bar_df, "gene_efficiency", iteration=self.iter_num,
                                has_converged=self.has_converged, unix_time=time.time())
         self.redis_db.publish(eta_bar_df, "gene_efficiency", iteration=self.iter_num,
@@ -405,14 +402,14 @@
         # prob = np.bincount(idx) / np.bincount(idx).sum()
         df = pd.DataFrame({
             'class_name': self.cellTypes.names,
             'counts': counts
         })
         self.redis_db.to_redis(df, "cell_type_posterior", iter_num=self.iter_num, has_converged=self.has_converged, unix_time=time.time())
         self.redis_db.publish(df, "cell_type_posterior", iteration=self.iter_num, has_converged=self.has_converged, unix_time=time.time())
-        logger.info("redis end")
+
```

## pciSeq/src/viewer/utils.py

```diff
@@ -26,27 +26,42 @@
     appDict = make_config_base(dst)
     cellBoundaries_tsv = os.path.join(dst, 'data', 'cellBoundaries.tsv')
     cellBoundaries_dict = {"mediaLink": "../../data/cellBoundaries.tsv",
                            "size": str(os.path.getsize(cellBoundaries_tsv))}
     roi_dict = {"x0": 0, "x1": w, "y0": 0, "y1": h}
     appDict['cellBoundaries'] = cellBoundaries_dict
     appDict['roi'] = roi_dict
-    appDict['zoomLevels'] = 10
-    appDict['tiles'] = "https://storage.googleapis.com/ca1-data/img/262144px/{z}/{y}/{x}.jpgZZZ"
+    appDict['maxZoom'] = 8
+    appDict['layers'] = {
+        'empty': "",
+        'dapi': "https://storage.googleapis.com/ca1-data/img/262144px/{z}/{y}/{x}.jpg",
+
+    }
+    appDict['spotSize'] = 1/16
 
     config_str = "// NOTES: \n" \
                  "// 1. paths are with respect to the location of 'streaming-tsv-parser.js \n" \
                  "// 2. roi is the image size in pixels. Leave x0 and y0 at zero and set x1 to the width and y1 to the height \n" \
-                 "// 3. tiles should point to the folder that keeps your pyramid of tiles. If you do not have that just \n" \
-                 "//    change the link to a blind one (change the jpg extension for example). The viewer should work \n" \
-                 "//    without the dapi background though \n" \
+                 "// 3. layers is a dict. Each key/value pair contains the string (the name) of the background image and the \n" \
+                 "//    location of the folder that the corresponding pyramid of tiles. If the tiles are stored locally, they \n" \
+                 "//    should be kept in a folder which is served, for example next to the tsv flatfiles. The path should be \n" \
+                 "//    in relation to the location of the index.html If you do not have a pyramid of tiles just \n" \
+                 "//    change the link to a blind one (change the jpg extension for example or just use an empty string). \n" \
+                 "//    The viewer should work without the dapi background though. \n" \
+                 "//    If the dict has more than one entries then a small control with radio button will appear at the top \n" \
+                 "//    right of the viewer to switch between different background images. \n" \
                  "// 4. size is the tsv size in bytes. I use os.path.getsize() to get it. Not crucial if you \n" \
                  "//    dont get it right, ie the full tsv will still be parsed despite this being wrong. It \n" \
                  "//    is used by the loading page piecharts to calc how far we are \n" \
-                 "// 5. Leave zoomLevels to 10 \n" \
+                 "// 5. maxZoom: maximum zoom levels. In most cases a value of 8 if good enough. If you have a big image, like \n" \
+                 "//    full coronal section for example then a value of 10 would make sense. Note that this should be typically \n" \
+                 "//    inline with the zoom level you used when you did \n" \
+                 "//    the pyramid of tiles. No harm is it less. If it is greater, then for these extra zoom levels there will \n" \
+                 "//    be no background image. \n" \
+                 "// 6. spotSize: Scalar. Use this to adjust the screen-size of your spots before they morph into glyphs. \n" \
                  " function config() { return %s }" % json.dumps(appDict)
     config = os.path.join(dst, 'viewer', 'js', 'config.js')
     with open(config, 'w') as data:
         data.write(str(config_str))
     logger.info(' viewer config saved at %s' % config)
```

## pciSeq/static/2D/viewer/js/config.js

### js-beautify {}

```diff
@@ -1,38 +1,48 @@
 // NOTES:
-// 1. roi is the image size in pixels. Leave x0 and y0 at zero and set x1 to the width and y1 to the height
-// 2. zoomLevels: maximum zoom levels.
-// 3. tiles should point to the folder that keeps your pyramid of tiles. If you do not have that just
-//    change the link to a blind one (use an empty string or change the jpg extension for example).
-//    The viewer should work without the dapi background.
-//    If the tiles are stored locally, they should be stored under the project root dir and the link
-//    should be the path from the project root dir too.
-// 4. cellData, geneData, cellBoundaries:
-//    4.1: mediaLink paths are with respect to the location of 'streaming-tsv-parser.js'
-//    4.2: size is the tsv size in bytes. I use os.path.getsize() to get it. Not crucial if you dont get it right,
-//         ie the full tsv will still be parsed despite this being wrong. It is used by the loading page
-//         donut charts to calc how far we are.
-
+// 1. paths are with respect to the location of 'streaming-tsv-parser.js
+// 2. roi is the image size in pixels. Leave x0 and y0 at zero and set x1 to the width and y1 to the height
+// 3. layers is a dict. Each key/value pair contains the string (the name) of the background image and the
+//    location of the folder that the corresponding pyramid of tiles. If the tiles are stored locally, they
+//    should be kept in a folder which is served, for example next to the tsv flatfiles. The path should be
+//    in relation to the location of the index.html If you do not have a pyramid of tiles just
+//    change the link to a blind one (change the jpg extension for example or just use an empty string).
+//    The viewer should work without the dapi background though.
+//    If the dict has more than one entries then a small control with radio button will appear at the top
+//    right of the viewer to switch between different background images.
+// 4. size is the tsv size in bytes. I use os.path.getsize() to get it. Not crucial if you
+//    dont get it right, ie the full tsv will still be parsed despite this being wrong. It
+//    is used by the loading page piecharts to calc how far we are
+// 5. maxZoom: maximum zoom levels. In most cases a value of 8 if good enough. If you have a big image, like
+//    full coronal section for example then a value of 10 would make sense. Note that this should be typically
+//    inline with the zoom level you used when you did
+//    the pyramid of tiles. No harm is it less. If it is greater, then for these extra zoom levels there will
+//    be no background image.
+// 6. spotSize: Scalar. Use this to adjust the screen-size of your spots before they morph into glyphs.
 function config() {
     return {
-        roi: {
+        "cellData": {
+            "mediaLink": "../../data/cellData.tsv",
+            "size": "2180603"
+        },
+        "geneData": {
+            "mediaLink": "../../data/geneData.tsv",
+            "size": "9630820"
+        },
+        "cellBoundaries": {
+            "mediaLink": "../../data/cellBoundaries.tsv",
+            "size": "1306209"
+        },
+        "roi": {
             "x0": 0,
             "x1": 7602,
             "y0": 0,
             "y1": 5471
         },
-        zoomLevels: 10,
-        tiles: 'https://storage.googleapis.com/ca1-data/img/262144px/{z}/{y}/{x}.jpg',
-        cellData: {
-            mediaLink: '../../data/cellData.tsv',
-            size: "2179232"
-        },
-        geneData: {
-            mediaLink: '../../data/geneData.tsv',
-            size: "9630856"
-        },
-        cellBoundaries: {
-            mediaLink: '../../data/cellBoundaries.tsv',
-            size: "1306209"
+        "maxZoom": 8,
+        "layers": {
+            // "empty": "",
+            "dapi": "https://storage.googleapis.com/ca1-data/img/262144px/{z}/{y}/{x}.jpg"
         },
+        "spotSize": 1 / 16
     }
 }
```

## pciSeq/static/2D/viewer/js/dapi.js

### js-beautify {}

```diff
@@ -1,49 +1,67 @@
 function dapi(cfg) {
     console.log('Doing Dapi plot');
 
-    var map_dims = mapSize(cfg.zoomLevels),
+    var map_dims = mapSize(cfg.maxZoom),
         tiles = cfg.tiles,
         roi = cfg.roi;
 
     var a = map_dims[0] / (roi.x1 - roi.x0),
         b = -map_dims[0] / (roi.x1 - roi.x0) * roi.x0,
         c = map_dims[1] / (roi.y1 - roi.y0),
         d = -map_dims[1] / (roi.y1 - roi.y0) * roi.y0;
 
     // This transformation maps a point from the roi domain to the domain defined by [0,0] and [someScalarX*256*2^maxZoom, someScalarY*256*2^maxZoom].
     // The scalar is there to make the bounding have the same shape as the roi.
     var t = new L.Transformation(a, b, c, d);
 
     // The transformation in this CRS maps the the top left corner to (0,0) and the bottom right to (256, 256)
-    // Leaflet thinks that the map is 256px-by-256px wide. These are the dimension of the tile at
-    // zoom = 0.
+    // Leaflet thinks that the map is 256px-by-256px wide. These are the dimension of the tile at zoom = 0.
     // Each side of the map however is 256 * 2 ** maxZoomLevel pixels wide.
     // For maxZoomLevel = 10 for example the map is 262144px-262144px
     // Hence we have to specify a factor of 256/262144 = 1/1024.
     // in general the factor is 256 / (256 * 2 ** maxZoomLevel)
-    var a_x = 256 / (256 * 2 ** cfg.zoomLevels),
-        c_y = 256 / (256 * 2 ** cfg.zoomLevels)
+    var a_x = 256 / (256 * 2 ** cfg.maxZoom),
+        c_y = 256 / (256 * 2 ** cfg.maxZoom)
     L.CRS.MySimple = L.extend({}, L.CRS.Simple, {
         transformation: new L.Transformation(a_x, 0, c_y, 0),
     });
 
     var southWest = L.latLng(map_dims[1], map_dims[0]),
         northEast = L.latLng(0, 0),
         mapBounds = L.latLngBounds(southWest, northEast);
 
     map = L.map('mymap', {
         crs: L.CRS.MySimple,
         attributionControl: false,
-    }).setView([map_dims[1] / 2, map_dims[0] / 2], 2);
-    L.tileLayer(tiles, {
         minZoom: 0,
-        maxZoom: 8,
+        maxZoom: cfg.maxZoom,
         bounds: mapBounds,
-    }).addTo(map);
+    }).setView([map_dims[1] / 2, map_dims[0] / 2], 2);
+
+
+    var baseLayers = {}
+    for (const [key, value] of Object.entries(cfg.layers)) {
+        baseLayers[key] = L.tileLayer(value);
+    }
+
+    var nLayers = Object.values(baseLayers).length
+    //Add control layers to map
+    if (nLayers > 1) {
+        L.control.layers(baseLayers, null, {
+            collapsed: false
+        }).addTo(map);
+    }
+
+    // It seems I can set the active layer, however on the control it looks as
+    // if the active one is the last layer. The bullet point always stays at
+    // the last layer
+    var selectedLayer = Object.values(baseLayers)[nLayers - 1]
+    // var selectedLayer = Object.values(baseLayers)[0] // that sets the top layer active, the bullet is still at the bottom!!!
+    selectedLayer.addTo(map)
 
     function getTaxonomy(gene) {
         if (glyphMap.get(gene)) {
             out = glyphMap.get(gene).taxonomy
         } else {
             out = glyphMap.get('Generic').taxonomy
         }
```

## pciSeq/static/2D/viewer/js/stage_markers_patched.js

### js-beautify {}

```diff
@@ -40,15 +40,16 @@
             );
             // Return the current iteration `result` value, this will be taken as next iteration `result` value and accumulate
             return result;
         }, {}); // empty object is the initial value for result object
     };
 
     function scaleRamp(z) {
-        var scale = 1 / 64;
+        // var scale = 1 / 64;
+        var scale = config().spotSize;
         return z === 0 ? scaleRampHelper(z, scale) :
             z === 1 ? scaleRampHelper(z, 2 * scale) :
             z === 2 ? scaleRampHelper(z, 2 * scale) : scaleRampHelper(z, 4 * scale)
 
         // return z === 0 ? 0.03 * 2**3 :
         //     z === 1 ? 0.03 * 2**3 :
         //         z === 2 ? 0.03 * 2**3 :
@@ -62,15 +63,15 @@
 
     function scaleRampHelper(z, scale) {
         // makes a tiny dot and then its scales it up based on the map and the dapi dimensions
         // As a general remark also, keep in mind that every time you zoom in, leaflet (I think) scales up by 2.
         // Divide by 2 to keep the marker the same as size. Hence if for zoom level = 3 the  return value from
         // this function is lets say zo 10, then when to keep the same size on the screen for the dot, at zoom = 4
         // the return value should be 5
-        var map_side = mapSide(configSettings.zoomLevels),
+        var map_side = mapSide(configSettings.maxZoom),
             dapi_size = [configSettings.roi.x1 - configSettings.roi.x0, configSettings.roi.y1 - configSettings.roi.y0],
             max_dapi = Math.max(...dapi_size),
             c = map_side / max_dapi,
             tiny_dot = 1 / (2 ** z),
             dot = c * tiny_dot;
         return dot * scale
     }
@@ -125,16 +126,16 @@
                 pc.baseTexture = texture.baseTexture;
             })
 
             // var innerContainer = geneContainer_array.filter(d => d.name === dummy_gene)[0];
             if (firstDraw) {
                 if (event.type === 'add') {
 
-                    initialScale = invScale / 8;
-                    initialScale = 0.125;
+                    // initialScale = invScale / 8;
+                    // initialScale = 0.125;
                     var targetScale = zoom <= zoomSwitch ? scaleRamp(zoom) : 1;
                     for (var i = 0; i < geneNames.length; i++) {
                         var gene = geneNames[i];
                         var innerContainer = geneContainer_array.filter(d => d.name === gene)[0];
                         innerContainer.localScale = targetScale;
                         var _data = data[gene];
                         for (var j = 0; j < _data.length; j++) {
```

## Comparing `pciSeq-0.0.52.dev1.dist-info/LICENCE` & `pciSeq-0.0.52rc0.dev0.dist-info/LICENCE`

 * *Files identical despite different names*

## Comparing `pciSeq-0.0.52.dev1.dist-info/METADATA` & `pciSeq-0.0.52rc0.dev0.dist-info/METADATA`

 * *Files 1% similar despite different names*

```diff
@@ -1,10 +1,10 @@
 Metadata-Version: 2.1
 Name: pciSeq
-Version: 0.0.52.dev1
+Version: 0.0.52rc0.dev0
 Summary: Probabilistic cell typing for spatial transcriptomics
 Home-page: https://github.com/acycliq/pciSeq
 Author: Dimitris Nicoloutsopoulos
 Author-email: dimitris.nicoloutsopoulos@gmail.com
 License: BSD
 Classifier: Programming Language :: Python :: 3
 Classifier: License :: OSI Approved :: BSD License
```

## Comparing `pciSeq-0.0.52.dev1.dist-info/RECORD` & `pciSeq-0.0.52rc0.dev0.dist-info/RECORD`

 * *Files 12% similar despite different names*

```diff
@@ -1,53 +1,54 @@
 pciSeq/__init__.py,sha256=mxqjE8smLqDQuwzheJ_iA3gfcyV7c4x8lHQPwSEQwJk,2135
 pciSeq/app.py,sha256=WTBF8N7hbVz6DE3yroUgXL23ophYgyQno_s5V-A0MF4,10910
 pciSeq/config.py,sha256=DFgt5LpDUBlLNQNecM-TfoTry3388LbdRU4jAxhxDHA,3367
 pciSeq/make_plot.py,sha256=2AkncrVVvytr2keK505oMC0LV1Txli9WpsSGpjmht_4,3678
 pciSeq/src/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
-pciSeq/src/_version.py,sha256=vEMUYFCvX9K7MEOv0lzZkZe_YDlbY9sfs-8nG0BqNSY,27
+pciSeq/src/_version.py,sha256=sbWsm0Hlx0ncNOGSZ5Po_YseZhVPUVhv0UwnE8epnec,29
 pciSeq/src/cell_call/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
-pciSeq/src/cell_call/datatypes.py,sha256=WxWGXn9rQ9pgvlBJtbl2RP2jIK4iEAKtw92AeyovNhk,20857
+pciSeq/src/cell_call/datatypes.py,sha256=om8kMxOUxKcAoQHSTgXEuX4UNMGtwTjFkOCGmeublqs,20864
 pciSeq/src/cell_call/log_config.py,sha256=vnm9TxlMEVG_GLPfl21GDEWTafEk3B-drTLsVQMfFEE,2995
-pciSeq/src/cell_call/main.py,sha256=CzF_XwgTojRL1Bke-f_nspapm-x9qGhW1ryu3g9thu0,19153
+pciSeq/src/cell_call/main.py,sha256=Vi4PvkIrljvyLpgK6AagXBdcreApX8uFkAqqbbVDQ7c,18989
 pciSeq/src/cell_call/summary.py,sha256=f2snLqJIeywtYuHrJLM-hRvifgqRU2u2aCKE_JrZJPs,3104
 pciSeq/src/cell_call/utils.py,sha256=M4KJrS82pIHuCoxg2ej6NE9fJAVCvJw90T7YmWI2T7o,9194
 pciSeq/src/diagnostics/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 pciSeq/src/diagnostics/config.py,sha256=kfPzvlFdGN0QgdfECwVV8lujXUiyim4fuetJoU4ziBQ,299
 pciSeq/src/diagnostics/diagnostics.py,sha256=L8tBgykN-KktrUS8YdIP5doi8meVVUUZU2PvOcUt72g,3336
 pciSeq/src/diagnostics/launch_diagnostics.py,sha256=p2t1wDc48iWvpfEjUt_-J5iK37EXPMg2O12y-pCeE_w,796
 pciSeq/src/diagnostics/utils.py,sha256=W77vAs1id1T47y7Wgt8VE9xut71TzgcN5ixXKEFPBMw,7162
 pciSeq/src/preprocess/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 pciSeq/src/preprocess/cell_borders.py,sha256=eTuFuexqGMFEV4upuTRInoaAi75NN46TIcvJ36TbZ4Q,11253
+pciSeq/src/preprocess/regionprops.py,sha256=gpScubR5adkKL5g9uSir8ud_pehiblMnDGc592c2S7c,5323
 pciSeq/src/preprocess/segmentation.py,sha256=9fc47EsxeFQnNtbJ0vm-RqATsBBMP73J5chdRFJocd4,6973
 pciSeq/src/preprocess/spot_labels.py,sha256=ukYTnp0xnu9J4-bKCTLEDhHAfaQNCvE7teEFyU0OLRk,4585
 pciSeq/src/preprocess/utils.py,sha256=7UB-fOKsSY_a4MVPs7E6BoNZcwITqm2N2JFMZqNYbj0,88
 pciSeq/src/viewer/__init__.py,sha256=47DEQpj8HBSa-_TImW-5JCeuQeRkm5NMpJWZG3hSuFU,0
 pciSeq/src/viewer/run_flask.py,sha256=9HS_RW-O9xddAeiApd4m_w2JJmbqUGZ5lhQ-z_HUeVA,2256
 pciSeq/src/viewer/stage_image.py,sha256=0zvdVk_R52T43ueSt3K3DgdMZPUyJNxDSyG0Bni7O78,5007
-pciSeq/src/viewer/utils.py,sha256=5jcauyJ2CKbcFxN4TTLVLsVlufAdqaLUQKXuZt6r7EU,7768
+pciSeq/src/viewer/utils.py,sha256=VwHLGIVQJnaRYeDAjoIoP87O8qXkeS7H4IHJynQPjsw,9109
 pciSeq/static/2D/index.html,sha256=9aMR5hOaXugJpSOt3ypUY24z6blsfOS83Cd9LHbAHa0,10895
 pciSeq/static/2D/viewer/genes_datatable.html,sha256=v10lTjkXYeAv0DVuaWHOB6XgG887k-_ABXAgobDV-tY,6506
 pciSeq/static/2D/viewer/css/index.css,sha256=IntuFc7kNAaNkLSL3m_9fpnH67i5KJpNWJ0OIW4TmD4,9622
 pciSeq/static/2D/viewer/css/progress.css,sha256=0MfjX9kIwD4wBeC39LBi9bCvu4ZkHwg_vtbuUiy4CuI,3135
 pciSeq/static/2D/viewer/js/classConfig.js,sha256=xhfWJHcU1bXipgJEokkMdUsrPDifPHpp_UHzW2qYopg,7435
-pciSeq/static/2D/viewer/js/config.js,sha256=1Eo75ruuB3USoaed3K8opHU3nh7y--Vi8UrkzNwQ4mE,1474
+pciSeq/static/2D/viewer/js/config.js,sha256=mhuRBtP_lLeEQzSWYb9D28C4dfufLu3KLFu2Edr0Zik,2314
 pciSeq/static/2D/viewer/js/customControl.js,sha256=niQd9FT0z_Dkp7v523n8plj7kNRnRpvADcHOChbDskY,5589
-pciSeq/static/2D/viewer/js/dapi.js,sha256=VgUvZ82pLg1w4k_nPuPP7oDDOCAKyRUUrfS3S5dtLlU,22811
+pciSeq/static/2D/viewer/js/dapi.js,sha256=m3y5z6GU2VQ7H74HKhDq_9RH0xNivt_TaZWM-HcNPDY,23501
 pciSeq/static/2D/viewer/js/dataLoader.js,sha256=iD3Kv9XNsSD2YyU3ilau5iaxAEQEX5VJnQ06VWyEP-g,5984
 pciSeq/static/2D/viewer/js/donut.js,sha256=dn9S3ZnNTFdEaiDU7ZadoKjVyvYp9LEJm82EvCe1ySY,17232
 pciSeq/static/2D/viewer/js/dt.js,sha256=eJpyKHNJjs1jf-GiTEa2_UsxMadO5eRWs2J0sJMhFms,4106
 pciSeq/static/2D/viewer/js/glyphConfig.js,sha256=23YX0DGc7XQXuJrUJDt59lref_TZPwUGF_Pr4h8c-tk,9835
 pciSeq/static/2D/viewer/js/glyphPaths.js,sha256=M3gwhPIRR17Q6cokLU754uovl5VwHQciduJKLlPhUXY,4389
 pciSeq/static/2D/viewer/js/glyphs.js,sha256=m6VFmUxNDJju9fXG9VzOKOnxip2mGbaBOyQKOY5_6zg,7224
 pciSeq/static/2D/viewer/js/index.js,sha256=IFMrGHYnC0fKzDgTfVlTrvLAqdNxlkjv_3_WNSAc7hw,12281
 pciSeq/static/2D/viewer/js/progress.js,sha256=781S0wMmKngJZxM7-opHPd3duQqlrdALLZZAwwACak8,1073
 pciSeq/static/2D/viewer/js/stage_cells.js,sha256=2rGYqLgUoMuvcI443YOPd9UiDTl85NP3GCQEUMEByPo,6367
 pciSeq/static/2D/viewer/js/stage_glyphs.js,sha256=egi2-UvAhO4aQ8vSOza_ld2opaJ2tclzqYqHneRnmmI,12064
 pciSeq/static/2D/viewer/js/stage_markers.js,sha256=nkEhmtk-8VGQbabDGK0fl4FURst9uUnKAGJcVGM9hcw,9540
-pciSeq/static/2D/viewer/js/stage_markers_patched.js,sha256=0DiG2kXvm5IloOa4n6oYgLTdaI8l5Sj5dsZWmwsCDhs,7534
+pciSeq/static/2D/viewer/js/stage_markers_patched.js,sha256=Cf5E_KuRkqYlwnDHBV84PaoH8fePxQxF-lMaBfTW7c4,7580
 pciSeq/static/2D/viewer/js/stage_polygons.js,sha256=G0Jxc7ocU2pNLKYtS-14_9jpGrFIR6RDiOyrOZ35PLI,15523
 pciSeq/static/2D/viewer/js/streaming-tsv-parser.js,sha256=ENPcWuYtmkUSDArHJ_1BnpnfJuFq-kpcOJSidE1I-Dw,4471
 pciSeq/static/2D/viewer/js/viewerUtils.js,sha256=609GVJUZY0P5wPTiPrgOZjeG_cmvm2jM7xze9Nl5H_0,4692
 pciSeq/static/2D/viewer/js/lib/css/L.Control.Layers.Tree.css,sha256=8M2Z28r2e7tj4G0eN52nlqGVunxHxMWHequekzF-fpo,995
 pciSeq/static/2D/viewer/js/lib/css/Leaflet.Coordinates-0.1.3.css,sha256=W6ZGjm-QrMQy_6C6xfkPA8tgk_Oh6JY3StLhQUrK_fA,421
 pciSeq/static/2D/viewer/js/lib/css/bootstrap.min.css,sha256=fO2Fh9Otx1Ft-Cy6-PgzCTeWj4fR-yJ7G9BrYgQNM9k,122544
 pciSeq/static/2D/viewer/js/lib/css/index.css,sha256=Tz2XH-M7Yg87oHCXT9T1v4WtfmVfZbxCKDgUMaTNgDU,8669
@@ -60,13 +61,13 @@
 pciSeq/static/2D/viewer/js/lib/js/leaflet-pip.js,sha256=QqsilU0SHTaKA_qNSv_jEDH53cWO6T_RdforfybnbKM,15374
 pciSeq/static/2D/viewer/js/lib/js/leaflet.textpath.js,sha256=i5cmMPqpqXShZz0TvXU3ZQUi2Y07SIbIpMYbhPE8gWE,5927
 pciSeq/static/2D/viewer/js/lib/js/preloader.js,sha256=eaFWRf9yT-Td4C2t01P8fDEZYszYTaGDgQ1vjevNIDI,265
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/MarkerContainer.js,sha256=Z8TADCqYEQr25OEF7wljsOVAOpx87UrjFPI2nvIty70,9097
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/bezier-easing.js,sha256=yMoRDqJiglg0M0eWM25WGA5zzLWuJpOyC3a8aGFx0jw,4434
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/example.min.js,sha256=bZEnK8XSnShrpLYtOS7Xq7wnp-nW_nMZh4aaDc1Gt6U,613808
 pciSeq/static/2D/viewer/js/lib/js/pixiOverlay/tools.min.js,sha256=A027Ub-oWXvKbz2B3pgMFS5xApmnEHXz9GQh79YZZnU,91874
-pciSeq-0.0.52.dev1.dist-info/LICENCE,sha256=6kbiFSfobTZ7beWiKnHpN902HgBx-Jzgcme0SvKqhKY,1091
-pciSeq-0.0.52.dev1.dist-info/METADATA,sha256=pkUDKEAKRvq4B2_mPn0KHGVumXYatmuoXHViKA5ESrI,4116
-pciSeq-0.0.52.dev1.dist-info/WHEEL,sha256=G16H4A3IeoQmnOrYV4ueZGKSjhipXx8zc8nu9FGlvMA,92
-pciSeq-0.0.52.dev1.dist-info/entry_points.txt,sha256=RGOkAd0S7rrTuEInj-jKu5B7630i4R3mK9v9vx6nqc8,48
-pciSeq-0.0.52.dev1.dist-info/top_level.txt,sha256=mSonMN0SaB51m5yLGm5hFE1Fthf-2U4sRKxSezY-JMk,7
-pciSeq-0.0.52.dev1.dist-info/RECORD,,
+pciSeq-0.0.52rc0.dev0.dist-info/LICENCE,sha256=6kbiFSfobTZ7beWiKnHpN902HgBx-Jzgcme0SvKqhKY,1091
+pciSeq-0.0.52rc0.dev0.dist-info/METADATA,sha256=EBiFhaJj2ful1qOoOvZoC5Ynnzt19YAh42TtTR8hEFc,4119
+pciSeq-0.0.52rc0.dev0.dist-info/WHEEL,sha256=G16H4A3IeoQmnOrYV4ueZGKSjhipXx8zc8nu9FGlvMA,92
+pciSeq-0.0.52rc0.dev0.dist-info/entry_points.txt,sha256=RGOkAd0S7rrTuEInj-jKu5B7630i4R3mK9v9vx6nqc8,48
+pciSeq-0.0.52rc0.dev0.dist-info/top_level.txt,sha256=mSonMN0SaB51m5yLGm5hFE1Fthf-2U4sRKxSezY-JMk,7
+pciSeq-0.0.52rc0.dev0.dist-info/RECORD,,
```

